### ISSUE TITLE
Exposing Client ID and Secret in Source Code

### ISSUE THREAD

**Issue by jaschaio** [infusionsoft/infusionsoft-php#83]
Hey there,
I have been wondering if its a problem if my Client ID and Client Secret are exposed in my application.

People that check the source code will find it.

It looks like the only way to request an Auth Token via the Infusionsoft API is passing along my Client ID and Secret. So there is no way to not use the Client ID and Secret if I want to use the OAuth Authentification Process.

The AWeber API Docs for example say the following regarding the Client ID and Secret:

> 
[…] you should never distribute your application's key / secret. This provides a means for other people to spoof your application and gain access to customer data under false pretenses, and may come back to damage the reputation of your application. Don't do it, ever. […]

They offer a different authentication method without Client ID and Secret.

So maybe someone can tell me if this is a problem and if there is an alternative.

Thank you and best regards

**Comment by micfai** [comment]
It sounds like you are using WordPress or some other similar platform. You need to use a proxy that handles the initial authentication and token refresh only.

**Comment by jaschaio** [comment]
Yes I use wordpress. I've already created a Proxy. Is there any authentification required between the plugin user and my proxy or is it enough that the authentification happens between the proxy and the Infusionsoft Authentification Server? I got negative comments on stackoverflow regarding my proxy approach: http://stackoverflow.com/questions/39029469/oauth-authentication-with-static-redirect-url-and-dynamic-client-url/39068300?noredirect=1#comment65551086_39068300
